sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -163,6 +163,24 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "v2 catalog doesn't support getCurrentDatabase yet, " +
+              "so the default database name cannot be deduced.")
+        case None =>
+          ShowTablesCommand(None, pattern)
+      }
+
+    case plan @ ShowTablesStatement(Some(namespace), pattern) =>
+      val CatalogObjectIdentifier(maybeCatalog, identifier) = namespace
+      val catalog = maybeCatalog.orElse(defaultCatalog)
+      catalog match {
+        case Some(v2Catalog) =>
+          ShowTables(plan.output, v2Catalog.asTableCatalog, identifier, pattern)
```

Comment:
I think it would be better to convert identifier to `Array[String]` here instead of in the exec node. It seems strange to use an Identifier when it is really a namespace.

Reply:
Changed.